core/src/main/java/org/apache/druid/timeline/VersionedIntervalTimeline.java

```diff
@@ -284,15 +286,15 @@ private void addAll(
 
   @Override
   @Nullable
-  public PartitionHolder<ObjectType> findEntry(Interval interval, VersionType version)
+  public PartitionChunk<ObjectType> findChunk(Interval interval, VersionType version, int partitionNum)
   {
     lock.readLock().lock();
     try {
       for (Entry<Interval, TreeMap<VersionType, TimelineEntry>> entry : allTimelineEntries.entrySet()) {
         if (entry.getKey().equals(interval) || entry.getKey().contains(interval)) {
           TimelineEntry foundEntry = entry.getValue().get(version);
           if (foundEntry != null) {
-            return foundEntry.getPartitionHolder().asImmutable();
```

Comment:
is the asImmutable method still needed?

Reply:
Looks like its not. will delete this method as well as the class ImmutablePartitionHolder.